src/cobra/core/reaction.py

```diff
@@ -1451,7 +1451,7 @@ def check_mass_balance(self) -> Dict[str, float]:
             for element, amount in metabolite.elements.items():
                 reaction_element_dict[element] += coefficient * amount
         # filter out 0 values
-        return {k: v for k, v in reaction_element_dict.items() if v != 0}
+        return {k: v for k, v in reaction_element_dict.items() if not isclose(v, 0.0, abs_tol=1e-12)}
```

Comment:
This is great. Maybe the tolerance could be taken from `Configuration().tolerance` instead like the other tolerances?